components/layout/ActivityCard.js

```diff
@@ -1,8 +1,18 @@
 import Link from "next/link";
 import Image from "next/image";
 import styled from "styled-components";
+import { FaHeart } from "react-icons/fa";
 
-export default function ActivityCard({ id, title, categories, imageUrl }) {
+export default function ActivityCard({
+  id,
+  title,
+  categories,
+  imageUrl,
+  bookmarks,
+  toggleBookmark,
+}) {
+  const isBookmarked = bookmarks.some((bookmark) => bookmark.id === id); 
+  
```

Comment:
Please review the suggested options for simplifying the bookmark logic.

By implementing one of these options, you can avoid the need for `ActivityCard` to calculate `isBookmarked` itself,  making the code cleaner, more efficient, and easier to maintain.